packages/gatsby/src/joi-schemas/joi.ts

```diff
@@ -32,6 +32,9 @@ export const gatsbyConfigSchema: Joi.ObjectSchema<IGatsbyConfig> = Joi.object()
           .replace(/^\/$/, ``)
       )
     ),
+    linkPrefix: Joi.forbidden().error(
+      new Error(`"linkPrefix" should be changed to "pathPrefix"`)
+    ),
```

Comment:
moved from https://github.com/gatsbyjs/gatsby/blob/874088c3a7e58f74b4538d95d77069c1c1cb9267/packages/gatsby/src/redux/reducers/config.js#L21-L23 to validation object, so validation is unified